pkg/ngrokapidriver/driver.go

```diff
@@ -56,27 +57,34 @@ func (nc ngrokAPIDriver) FindEdge(ctx context.Context, id string) (*ngrok.HTTPSE
 // * tunnel group backends
 // * edge routes
 // * the edge itself
-func (napi ngrokAPIDriver) CreateEdge(ctx context.Context, edgeSummary Edge) (*ngrok.HTTPSEdge, error) {
+func (napi ngrokAPIDriver) CreateEdge(ctx context.Context, edgeSummary *Edge) (*ngrok.HTTPSEdge, error) {
 	log := ctrl.LoggerFrom(ctx)
 	// TODO: Support multiple rules and multiple hostports
-	domain := strings.Split(edgeSummary.Hostport, ":")[0]
-	_, err := napi.reservedDomains.Create(ctx, &ngrok.ReservedDomainCreate{
+	domain, _, err := net.SplitHostPort(edgeSummary.Hostport)
+	if err != nil {
+		return nil, err
+	}
+	_, err = napi.reservedDomains.Create(ctx, &ngrok.ReservedDomainCreate{
 		Name:        domain,
 		Region:      "us", // TODO: Set this from user config
 		Description: "Created by ngrok-ingress-controller",
 		Metadata:    napi.metadata,
 	})
 	// Swallow conflicts, just always try to create it and don't delete them upon ingress deletion
 	if err != nil {
-		if strings.Contains(err.Error(), "ERR_NGROK_413") || strings.Contains(err.Error(), "ERR_NGROK_7122") {
-			log.Info("Reserved domain already exists, skipping creation", "domain", domain)
-		} else {
-			return nil, err
+		var nerr *ngrok.Error
+		if errors.As(err, &nerr) {
```

Comment:
Using `ngrok.Error.ErrorCode` is better than substring matching on the whole error text.